wrappers/rust/icicle-core/src/field.rs

```diff
@@ -15,6 +15,9 @@ pub struct Field<const NUM_LIMBS: usize, F: FieldConfig> {
     p: PhantomData<F>,
 }
 
+unsafe impl<const NUM_LIMBS: usize, F: FieldConfig> Send for Field<NUM_LIMBS, F> {}
+unsafe impl<const NUM_LIMBS: usize, F: FieldConfig> Sync for Field<NUM_LIMBS, F> {}
```

Comment:
What are these for? Field should derive Send and Sync automatically, doesn't it?

Reply:
I just tested it on my machine - it works without send and sync

Reply:
umm right as we discussed on a meeting - as long as you don't pass Field across the threads - all is fine, but if you do generate scalars on one thread and reuse across other threads as unmodified params for example etc - then you can't without these - automatic derive is prevented. I didn't add tests with that scenario because current tests cover that and make it a bit redundant but I didn't  want to modify existing tests as well  - again imo it's a bit redundant as they look like twins except inputs reusing across threads - and basic tutorial for `Arc(Mutex())`

Reply:
I see. Ok then